### ISSUE TITLE
Feeling dumb :(

### ISSUE THREAD

**Issue by jordanpapaleo** [lerna/lerna-changelog#87]
Hi - 

I globally installed lerna-changelog:

![image](https://user-images.githubusercontent.com/1728807/33471323-a628dca0-d621-11e7-89ed-06de6f6f8f23.png)

I setup my config:

```json
{
  "lerna": "2.0.0",
  "version": "0.7.0",
  "ignoreCommitters": false,
  "changelog": {
    "repo": "clearcapital/ccp",
    "labels": {
      "feature": ":rocket: New Feature",
      "bug": ":bug: Bug Fix"
    },
    "cacheDir": ".changelog"
  }
}
```
I verified that I am using correct labels for labels in my repo.

![image](https://user-images.githubusercontent.com/1728807/33471271-60fb4366-d621-11e7-99e4-ce0827502949.png)

![image](https://user-images.githubusercontent.com/1728807/33471278-66dd9158-d621-11e7-9d95-43a421357913.png)

I created my GitHub API access token selecting the repo option as I have a private repo.  I used that key to authenticate.

When I run `lerna-changelog`:

![image](https://user-images.githubusercontent.com/1728807/33471374-e83bb4be-d621-11e7-9e05-492d8d3ac6a3.png)

The cacheDir at my application root is created but nothing to 'CopyPasta' in the console.  I feel like I am missing something really dumb here.

How do I get the output to copy into a changelog.md file?

Thanks!.

Jordan

**Comment by emmenko** [comment]
The setup looks good, you just need to have some releases/tags first, otherwise there is nothing to be shown 😉

You can see below all the steps necessary to gather the list of releases:
https://github.com/lerna/lerna-changelog/blob/bcc8a3a6e9445c69c5713183c86708b97cbcbb3a/src/changelog.ts#L36-L66

**Comment by jordanpapaleo** [comment]
Good Morning!

So isnt this supposed to use the labels applied in GitHub?

`Uses github PR/Issue names categorized by labels with configurable headings`

Here is an example from my closed issues on my project:

![image](https://user-images.githubusercontent.com/1728807/33492445-eb6bb2c4-d671-11e7-86d6-b3a67f3bfc2a.png)

Here are tags from my git tags:

```bash
release/0.2.0-mba
v0.1.1
v0.1.2
v0.1.3
v0.10.0
v0.10.1
v0.10.3
v0.11.0
v0.12.0
v0.13.0
v0.14.0
v0.14.1
v0.14.2
v0.15.0
v0.16.0
v0.16.1
v0.2.0
v0.3.0
v0.4.0
v0.5.0
v0.6.0
v0.7.0
v0.7.1
v0.8.0
v0.9.0
v0.9.1
v0.9.2
v0.9.3
```

I am super pumped on this project and hope we can figure this out.

**Comment by emmenko** [comment]
Ah I see. I guess at this point the easiest and quickest way to find out where the problem is is to debug through the steps I linked above.

Simply add some logs into the `node_modules/lerna-changelog` file and see where things are missing 🕵️

Thanks

**Comment by dannyphillips** [comment]
@jordanpapaleo I'm having the same issue. Were you able to figure out what went wrong?

**Comment by jordanpapaleo** [comment]
No. I have not had the chance to debug the lerna code but it’s on my list.   I’ll post whatever I find out.  Will you do the same?

I love to see a step by step showing it work.

**Comment by taicho** [comment]
Any luck on this? I can't for the life of me figure out how to get lerna-changelog to product any output whatsover...

### lerna.json
```
  "lerna": "2.8.0",
  "packages": [
    "packages/*"
  ],
  "version": "0.0.3",
  "changelog": {
    "repo": "myorg/myrepo",
    "labels": {
      "enhancement": ":rocket: Enhancement",
      "bug": ":bug: Bug Fix"
    },
    "cacheDir": ".changelog"
  }
}
```

I've tried doing commits then lerna-changelog, branching then doing commits which fix/reference issues with labels then doing a PR and merging. All with no output generated, am I missing something, am I doing something wrong?

**Comment by jordanpapaleo** [comment]
Is it a private repo?  I think my issue might be with correctly authenticating. 

But I have made no progress on it yet.

**Comment by taicho** [comment]
@jordanpapaleo It is indeed a private repo. That being said, I authenticated using a personal token (with full repo permissions)  ala GITHUB_AUTH and it appears that it even downloads stuff into the .changelog directory but no output or anything beyond that.

**Comment by taicho** [comment]
@jordanpapaleo I ended up switching to [https://github.com/skywinder/github-changelog-generator](https://github.com/skywinder/github-changelog-generator)... I really liked the style of this changelog but with the stack of issues, a PR on definite hold and 4 months since the last code change it's starting to feel like this repo/package has stagnated (at least for the time being).

**Comment by jordanpapaleo** [comment]
I was using my personal token as well with no success. I’ll check that out.

**Comment by Turbo87** [comment]
@taicho can you provide a public repo that reproduces the issue? `lerna-changelog` is working fine for me and I'm curious what the issue is.

the regular workflow is:
- commit code as pull request
- apply labels to the pull request
- merge the pull request
- use lerna-changelog to generate changelog from labels pull requests

Issues are not included, only PRs.

**Comment by inakianduaga** [comment]
I also cannot get any output, and the project is public

https://github.com/Scout24/as24-custom-events

so definitely something missing / broken

**Comment by Turbo87** [comment]
> So definitely the error display on rejected github API calls should be improved (it was a 403).

PRs welcome :)

**Comment by suanmei** [comment]
Meet with the problem, but I have solved it。

There are a few points to note：

1. After you merge the pull request that already has a label added，you should checkout **master** branch locally

2. Maybe the issue request is a failure，for me ，it reported  `Bad credentials` , so I update my GITHUB_AUTH and finally it succeeded
